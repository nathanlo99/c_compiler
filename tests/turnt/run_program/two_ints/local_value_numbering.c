
int wain(int a, int b) {
  int c = 0;
  b = 2;
  c = a + b;
  a = 100;
  b = 50 + 50;
  return a + b + c;
}
