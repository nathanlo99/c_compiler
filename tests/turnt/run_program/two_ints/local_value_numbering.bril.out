@wain(a: int, b: int) : int {
  c: int = const 0;
  _t0: int = const 2;
  b: int = id _t0;
  _t1: int = id b;
  _t2: int = id a;
  _t3: int = id b;
  _t4: int = add _t2 _t3;
  c: int = id _t4;
  _t5: int = id c;
  _t6: int = const 100;
  a: int = id _t6;
  _t7: int = id a;
  _t8: int = const 50;
  _t9: int = const 50;
  _t10: int = add _t8 _t9;
  b: int = id _t10;
  _t11: int = id b;
  _t12: int = id a;
  _t13: int = id b;
  _t14: int = add _t12 _t13;
  _t15: int = id c;
  _t16: int = add _t14 _t15;
  ret _t16;
}

